tests/spec/core/inlines-spec.js

```diff
@@ -435,4 +435,19 @@ describe("Core - Inlines", () => {
       "#dom-referrerpolicy-no-referrer"
     );
   });
+
+  it("checks data-link-type of inline citations", async () => {
```

Comment:
Might be better to add these tests within: https://github.com/w3c/respec/blob/6a18cf58d1e9853e92344753983ec9502d09caeb/tests/spec/core/inlines-spec.js#L38-L47

A single test like: `expect(links[0].dataset.linkType).toBe("biblio")` should be enough.